### PR TITLE
docs: Add troubleshooting docs for Ingress

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to add troubleshooting steps for Ingress resources, the
similar step can be performed for CiliumClusterwideEnvoyConfig and
CiliumEnvoyConfig resources as well. I have listed down the common issue
and how to check for each request hoop, will improve along the way.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

### CODE REVIEW

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        NAME                           TYPE           CLUSTER-IP     EXTERNAL-IP    PORT(S)        AGE
+        cilium-ingress-basic-ingress   LoadBalancer   10.97.60.117   10.97.60.117   80:31911/TCP   17m
+
+        $ kubectl get cec cilium-ingress-default-basic-ingress
+        NAME                                   AGE
+        cilium-ingress-default-basic-ingress   18m
+
+ #. Validate if the Load Balancer service is having either external IP or FQDN assigned
+    If it's not available after long time, please check the Load Balancer related
+    documentation from your respective cloud provider.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
 #. Validate that the Load Balancer service has either an external IP or FQDN assigned.
    If it's not available after a long time, please check the Load Balancer related
    documentation from your respective cloud provider.
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+ #. Validate if there is any warning or error message while Cilium is trying to provision
+    the ``CiliumEnvoyConfig`` resource. This is unlikely to happen for CEC resources
+    originating from the Cilium Ingress controller.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
 #. Check if there is any warning or error message while Cilium is trying to provision
    the ``CiliumEnvoyConfig`` resource. This is unlikely to happen for CEC resources
    originating from the Cilium Ingress controller.
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    .. include:: ../gettingstarted/servicemesh/warning.rst
+
+
+Connectivity Troubleshooting
+----------------------------
+
+This section is for troubleshooting connectivity issues mainly for Ingress resources, but
+the same steps can be applied to manually configured CiliumEnvoyConfig resources as well.
+
+It's the best to have ``debug`` and ``debug-verbose`` enabled with below values. Kindly
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
It's best to have ``debug`` and ``debug-verbose`` enabled with below values. Kindly
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    .. code-block:: shell-session
+
+        $ kubectl get -n kube-system cm cilium-config -o json | grep "debug"
+                "debug": "true",
+                "debug-verbose": "flow",
+
+.. note::
+
+    The Ingress traffic is always allowed to pass through Cilium, regardless of the of
+    the related CiliumNetworkPolicy for underlying pods.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
    The Ingress traffic is always allowed to pass through Cilium, regardless of
    the related CiliumNetworkPolicy for underlying pods.
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+
+The request is normally traversing from LoadBalancer service to pre-assigned port of your
+node, then get forwarded to the Cilium Envoy proxy, and finally get proxied to the actual
+backend service.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
The request normally traverses from LoadBalancer service to pre-assigned port of your
node, then gets forwarded to the Cilium Envoy proxy, and finally gets proxied to the actual
backend service.
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+
+ #. The first step between cloud Load Balancer to node port is out of Cilium scope. Please
+    check related documentation from your respective cloud provider to make sure your
+    clusters are configured properly.
+
+ #. The second step could be checked by ssh to your underlying host, and send the similar
+    request to localhost.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
 #. The second step could be checked by connecting with SSH to your underlying host, and sending a
    request to localhost on the relevant port:
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+    .. code-block:: shell-session
+
+        $ kubectl -n kube-system get cm cilium-config -o json | egrep "enable-ingress-controller|enable-envoy-config"
+                "enable-envoy-config": "true",
+                "enable-ingress-controller": "true",
+
+Ingress Troubleshooting
+-----------------------
+
+Internally, the Cilium Ingress controller will create one Load Balancer service, one CiliumEnvoyConfig
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
Internally, the Cilium Ingress controller will create one Load Balancer service, one ``CiliumEnvoyConfig``
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+    originating from the Cilium Ingress controller.
+
+    .. include:: ../gettingstarted/servicemesh/warning.rst
+
+
+Connectivity Troubleshooting
+----------------------------
+
+This section is for troubleshooting connectivity issues mainly for Ingress resources, but
+the same steps can be applied to manually configured CiliumEnvoyConfig resources as well.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
the same steps can be applied to manually configured ``CiliumEnvoyConfig`` resources as well.
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+        < date: Thu, 07 Jul 2022 12:25:56 GMT
+        < server: envoy
+        <
+        * Connection #0 to host localhost left intact
+
+        # Flows for world identity
+        $ kubectl -n kube-system exec ds/cilium -- hubble observe -f --identity 2
+        Jul  7 12:28:27.970: 127.0.0.1:54704 <- 127.0.0.1:13681 http-response FORWARDED (HTTP/1.1 503 0ms (GET http://localhost:31911/))
+
+    Alternatively, you can also send request directly to Envoy proxy port. For Ingress, the proxy port is randomly
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
    Alternatively, you can also send a request directly to the Envoy proxy port. For Ingress, the proxy port is randomly
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+    assigned by the Cilium Ingress controller. For manually configured ``CiliumEnvoyConfig`` resources, the proxy
+    port is retrieved directly from the spec.
+
+    .. code-block:: shell-session
+
+        $  kubectl logs -f -n kube-system ds/cilium --timestamps| egrep "envoy|proxy"
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
        $  kubectl logs -f -n kube-system ds/cilium --timestamps | egrep "envoy|proxy"
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+        > Accept: */*
+        >
+        * Mark bundle as not supporting multiuse
+        < HTTP/1.1 503 Service Unavailable
+        < content-length: 19
+        < content-type: text/plain
+        < date: Fri, 08 Jul 2022 08:12:35 GMT
+        < server: envoy
+
+    If you see the response similar to above, it means that the request is being
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
    If you see a response similar to the above, it means that the request is
```

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+
+        $ kubectl exec -n kube-system ds/cilium -- cilium status
+        ...
+        Controller Status:       49/49 healthy
+        Proxy Status:            OK, ip 10.0.0.25, 6 redirects active on ports 10000-20000
+        Global Identity Range:   min 256, max 65535
+
+ #. Assume that the above steps are done successfully, you can either send requests either
+    via external IP/FQDN In order to check if the request is proxied to backend service
+    correctly.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]: “Assuming”, and this sentence is not correct, please fix it

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+
+    Please double check if your backend service is up and healthy. The Envoy Discovery Service
+    (EDS) is having name following the convention ``<namespace>/<service-name>:<port>``.
```
Comment (qmonnet) [inline Documentation/operations/troubleshooting_servicemesh.rst]:
```suggestion
    Double-check whether your backend service is up and healthy. The Envoy Discovery Service
    (EDS) has a name that follows the convention ``<namespace>/<service-name>:<port>``.
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks! The documentation looks good, I've got some minor comments on the grammar.

I note this is a long addition to the troubleshooting page, maybe we should split it at some point. But this is beyond the scope of your PR.

**Comment by sayboras** [comment]
Request reviews from others, who might have more insights.

File: Documentation/operations/troubleshooting_servicemesh.rst [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+
+ #. Assuming that the above steps are done successfully, you can proceed to send a request via
+    an external IP or via FQDN next.
+
+    Double-check whether your backend service is up and healthy. The Envoy Discovery Service
+    (EDS) has a name that follows the convention ``<namespace>/<service-name>:<port>``.
+
+    .. code-block:: shell-session
+
+        $ LB_IP=$(kg ingress basic-ingress -o json | jq '.status.loadBalancer.ingress[0].ip' | jq -r .)
```
Comment (youssefazrak) [inline Documentation/operations/troubleshooting_servicemesh.rst]: small nit: `kg` --> `kubectl get`

**Review by qmonnet — APPROVED** [review]
Thanks!

**Pull request merged**